integration-testing/src/test/java/org/triplea/http/server/LobbyWebsocketClientTest.java

```diff
@@ -0,0 +1,53 @@
+package org.triplea.http.server;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import org.java_websocket.client.WebSocketClient;
+import org.java_websocket.handshake.ServerHandshake;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.triplea.http.client.lobby.chat.LobbyChatClient;
+import org.triplea.server.http.DropwizardTest;
+
+class LobbyWebsocketClientTest extends DropwizardTest {
+
+  @Test
+  @DisplayName("Verify basic websocket operations: open, send, close")
+  void verifyConnectivity() throws Exception {
+    final URI websocketUri = URI.create(localhost + LobbyChatClient.WEBSOCKET_PATH);
+
+    final WebSocketClient client =
+        new WebSocketClient(websocketUri) {
+          @Override
+          public void onOpen(final ServerHandshake serverHandshake) {}
+
+          @Override
+          public void onMessage(final String message) {}
+
+          @Override
+          public void onClose(final int code, final String reason, final boolean remote) {}
+
+          @Override
+          public void onError(final Exception ex) {}
+        };
+
+    assertThat(client.isOpen(), is(false));
+    client.connect();
+
+    await().pollDelay(10, TimeUnit.MILLISECONDS).atMost(1, TimeUnit.SECONDS).until(client::isOpen);
+    client.send("sending! Just to make sure there are no exception here.");
+
+    // small wait to process any responses
+    Thread.sleep(10);
+
+    client.close();
+    await()
+        .pollDelay(10, TimeUnit.MILLISECONDS)
+        .atMost(100, TimeUnit.MILLISECONDS)
+        .until(() -> !client.isOpen());
```

Comment:
I mean I get that we're just testing a library here, but in case any those tests ever fail, what would be do about it?
Probably nothing, because library change is too much and it worked in the past, so there must be a workaround

Reply:
This was one of the very first tests written before any websocket code landed, probably good to keep that context in mind. If for any reason the websocket connection does not open, is not configured right, or any of the flow with the wiring for message service is broken, we're verifying that level of connectivity. This is in another sense an integration test that does not rely on the `GenericWebsocketClient`, so if we see any issues we know there is something wrong on the server side. Hence, it's not really a library test, we are firing up a server, opening ports, and using the most basic client to connect to it. Perhaps the name `ClientTest` is misleading, this is a test using a `WebsocketClient` connecting to the lobby.